### ISSUE TITLE
Index was outside the bounds

### ISSUE THREAD

**Issue by lastowl** [cureos/csj2k#15]
The attached sample image cant be decompressed, it does open in irfanview alright it happens with any image which is compressed with 33003 jp2k
[file15.zip](https://github.com/cureos/csj2k/files/1266226/file15.zip)

The exact error it get is 'Index was outside the bounds of the array. at CSJ2K.j2k.image.ImgDataConverter.getData(DataBlk blk, Int32 c, Boolean intern) '

**Comment by anders9ustafsson** [comment]
Thanks for reporting, @lastowl. Which version of *csj2k* are you using?

**Comment by lastowl** [comment]
I've tried with the latest master and with nuget v2.0.0.1

**Comment by lastowl** [comment]
I also tried the the prerelease 3 on nuget same issue, if you need anymore images I can provide as manay as required

**Comment by anders9ustafsson** [comment]
@lastowl Thanks! I have not had time to look at this issue yet, I hope to be able to next month.

**Comment by SylvainDebaudringhien** [comment]
Hi. I've checked the image, and xnView cannot decode it as well.
Just for information... Sylvain.

**Comment by boxerab** [comment]
This file is a raw code stream, by the way, not in JP2 format. Perhaps this is causing the problem.

**Comment by lastowl** [comment]
Yeah the file is an exported tile from a tiff image, I’ve used the same method for a different compression on a grey scale j2k image and its alright just not this compression type.
